GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+
+### Procedures
+
+`Procedures` define the rule according to which votes are run. There can only be one active procedure at any given time. If governance wants to change a procedure, either to modify a value or add/remove a parameter, a new procedure has to be created and the previous one rendered inactive.
+
+```Go
+type Procedure struct {
+  VotingPeriod      uint64              //  Length of the voting period. Initial value: 2 weeks
+  MinDeposit        uint64              //  Minimum deposit for a proposal to enter voting period. 
+  OptionSet         []string            //  Options available to voters. {Yes, No, NoWithVeto, Abstain}
+  ProposalTypes     []string            //  Types available to submitters. {PlainTextProposal, SoftwareUpgradeProposal}
+  Threshold         uint64              //  Minimum value of Yes votes to No votes ratio for proposal to pass. Initial value: 0.5
+  Veto              rational.Rational   //  Minimum value of Veto votes to Total votes ratio for proposal to be vetoed. Initial value: 1/3
+  MaxDepositPeriod  uint64              //  Maximum period for Atom holders to deposit on a proposal. Initial value: 2 months
+  GovernancePenalty uint64              //  Penalty if validator does not vote
+  
+  ProcedureNumber   uint16              //  Incremented each time a new procedure is created
+  IsActive          bool                //  If true, procedure is active. Only one procedure can have isActive true.
+}
+```
+
+### Proposals
+
+`Proposals` are item to be voted on. They can be submitted by any Atom holder via a `TxSubmitProposal` transaction.
+
+```Go
+type TxSubmitProposal struct {
+  Title           string        //  Title of the proposal
+  Description     string        //  Description of the proposal
+  Type            string        //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category        bool          //  false=regular, true=urgent
+  InitialDeposit  uint64        //  Initial deposit paid by sender. Must be strictly positive.
+}
+
+type Proposal struct {
+  Title             string              //  Title of the proposal
+  Description       string              //  Description of the proposal
+  Type              string              //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category          bool                //  false=regular, true=urgent
+  Deposit           uint64              //  Current deposit on this proposal. Initial value is set at InitialDeposit
+  SubmitBlock       uint64              //  Height of the block where TxSubmitProposal was included
+  VotingStartBlock  uint64              //  Height of the block where MinDeposit was reached. -1 if MinDeposit is not reached.
+  Votes             map[string]uint64   //  Votes for each option (Yes, No, NoWithVeto, Abstain)
+}
+```
+
+Each `Proposal` is identified by its unique `proposalID`. 
```

Comment:
is the proposalID the key in the KVStore? I noticed it's not included in the `proposal` struct... We should add discussion surrounding indexing and the governance state... talk to bucky about organization of the specs

Reply:
Yes for me `ProposalID` is the key. @sunnya97 is doing the implementation, so he should lead the discussion